lib/charms/archive_auth_mirror/repository.py

```diff
@@ -44,3 +48,21 @@ def split_repository_uri(uri):
     """Split the repository URI into components."""
     parts = ('url', 'suite', 'components')
     return dict(zip(parts, uri.split(' ', maxsplit=2)))
+
+
+def update_config(get_paths=get_paths, suite=None, sign_key_id=None,
+                  new_ssh_peers=None):
```

Comment:
I'd move get_paths as last argument, since it's only meant to be passed in tests